### DISCUSSION TITLE
Is Ganache compatible with node 12?

### DISCUSSION THREAD

**Discussion by BarrySauce** [ConsenSys-archive/ganache#2875] [Q&A] [answered]
I am currently using node 12.22.11, may I know if Ganache is compatible with this node version?
When I activate Ganache or Ganache-Cli in my terminal, there would be error occuring:
<img width="1512" alt="Screenshot 2022-04-11 at 11 15 54 AM" src="https://user-images.githubusercontent.com/84897434/162659401-015e7669-46d0-4ca3-87ed-f2dd7894414c.png">

**Comment by davidmurdoch** [comment, marked as answer]
Yes, we currently test Node 12.0.0, 14, 16, and 17. What OS and CPU architecture are you running?

You should note that Node 12 will reach its end of life this month, and Ganache will drop Node 12 testing after May 31st. You should migrate to a newer version of Node ASAP.

**Reply by BarrySauce** [reply]
Hi sir, thank you for your reply! I am currently using MacOS, the chip is Apple M1 Pro, so I believe it is using ARM architecture. I will switch to node 12.0.0 and give it a try, since I am using another tool together with Ganache and it has specific requirement for node 12.

**Reply by davidmurdoch** [reply]
I wouldn't use node 12.0.0, you should use the latest node 12 version. What tool are you using that still requires node 12?

**Reply by BarrySauce** [reply]
Hi sir, I tried the node 12.0.0 version, Ganache can run on it properly now. And the tool I am using is Etheno, you can refer to their repo: https://github.com/crytic/etheno
Basically it's a track tool that can capture the transactions happening on Ganache :)

**Reply by davidmurdoch** [reply]
etheno is a python tool; are you sure you can't use a more up-to-date node version, like v16?

**Reply by BarrySauce** [reply]
Yes, in their repo they recommend to use Etheno with node 12. Now Ganache can be activated directly in node 12.0.0, but there is still error when Ganache is used with Etheno (which I believe is due to Etheno), I will switch to node 16 and try it again, thanks again for your help :)

**Reply by davidmurdoch** [reply]
That warning is due to an old bug that has since been fixed.

**Reply by BarrySauce** [reply]
Yes sir. I tried node 16 and Ganache works correctly now with Etheno. Thank you for your reminder :)
